crates/cli/src/subcommands/publish.rs

```diff
@@ -198,7 +199,12 @@ pub async fn exec(config: Config, args: &ArgMatches) -> Result<(), anyhow::Error
             ));
         }
         PublishResult::PermissionDenied { domain } => {
-            let identity = decode_identity(&config)?;
+            if anon_identity {
+                anyhow::bail!("You need to be logged in to publish to {}", domain.tld());
+            }
```

Comment:
We used to just rely on the error from `decode_identity` that said something like "you're not logged in, please log in". That somewhat obfuscated that we had a `PermissionDenied` error, so now we emit a more descriptive error.

(This was necessary because our new helper function `get_login_token_or_log_in` actually _logs the user in_, which is a very weird UX to have in an error case, especially when the user passed `--anon-identity`.)